llvm/lib/Transforms/IPO/DeadArgumentElimination.cpp

```diff
@@ -714,6 +758,83 @@ void DeadArgumentEliminationPass::PropagateLiveness(const RetOrArg &RA) {
   Uses.erase(Begin, I);
 }
 
+// Update kernel arguments table inside the integration header.
+// For example:
+//   static constexpr const bool param_omit_table[] = {
+//     // OMIT_TABLE_BEGIN
+//     // kernel_name_1
+//     false, false,     // <= update to true if the argument is dead
+//     // kernel_name_2
+//     false, false,
+//     // OMIT_TABLE_END
+//   };
+//   TODO: batch changes to multiple SYCL kernels and do one bulk update.
+constexpr StringLiteral OMIT_TABLE_BEGIN("// OMIT_TABLE_BEGIN");
+constexpr StringLiteral OMIT_TABLE_END("// OMIT_TABLE_END");
+static void updateIntegrationHeader(StringRef SyclKernelName,
```

Comment:
Looks OK...hopefully the header file is short and/or the number of kernels is small, as we are writing the whole file on each update. Otherwise we have to go with a "F F F F" kind of table that can be modified in-place without changing the size.

Reply:
I don't think it will make a difference in practice. But I'm working on a patch for doing a bulk update of int-header. I would like to do this change incrementally, i.e. in a separate patch.